etl/src/v2/state/store/postgres.rs

```diff
@@ -0,0 +1,269 @@
+#[sqlx(type_name = "etl.table_state", rename_all = "snake_case")]
+pub enum TableState {
+    Init,
+    DataSync,
+    FinishedCopy,
+    SyncDone,
+    Ready,
+    Skipped,
+}
+
+#[derive(Debug, Error)]
+pub enum ToTableStateError {
+    #[error("In-memory table replication phase can't be saved in the state store")]
+    InMemoryPhase,
+}
+
+impl TryFrom<TableReplicationPhase> for TableState {
+    type Error = ToTableStateError;
+
+    fn try_from(value: TableReplicationPhase) -> Result<Self, Self::Error> {
+        Ok(match value {
+            TableReplicationPhase::Init => TableState::Init,
+            TableReplicationPhase::DataSync => TableState::DataSync,
+            TableReplicationPhase::FinishedCopy => TableState::FinishedCopy,
+            TableReplicationPhase::SyncDone { .. } => TableState::SyncDone,
+            TableReplicationPhase::Ready => TableState::Ready,
+            TableReplicationPhase::Skipped => TableState::Skipped,
+            _ => return Err(ToTableStateError::InMemoryPhase),
+        })
+    }
+}
+
+#[derive(Debug, FromRow)]
+pub struct ReplicationStateRow {
+    pub pipeline_id: i64,
+    pub table_id: SqlxOid,
+    pub state: TableState,
+}
+
+#[derive(Debug)]
+struct Inner {
+    table_states: HashMap<TableId, TableReplicationPhase>,
+}
+
+#[derive(Debug, Clone)]
+pub struct PostgresStateStore {
+    pipeline_id: PipelineId,
+    source_config: SourceConfig,
+    inner: Arc<RwLock<Inner>>,
+}
+
+impl PostgresStateStore {
+    pub fn new(pipeline_id: PipelineId, source_config: SourceConfig) -> PostgresStateStore {
+        let inner = Inner {
+            table_states: HashMap::new(),
+        };
+        PostgresStateStore {
+            pipeline_id,
+            source_config,
+            inner: Arc::new(RwLock::new(inner)),
+        }
+    }
+
+    async fn connect_to_source(&self) -> Result<PgPool, sqlx::Error> {
+        let SourceConfig {
+            host,
+            port,
+            name,
+            username,
+            password,
+            tls,
+        } = &self.source_config;
+        let options = PgConnectOptions::new()
+            .application_name("replicator_migrator")
+            .host(host)
+            .port(*port)
+            .username(username)
+            .database(name);
+        let options = if let Some(password) = password {
+            options.password(password)
+        } else {
+            options
+        };
+        let TlsConfig {
+            trusted_root_certs,
+            enabled: tls_enabled,
+        } = tls;
+
+        let options = if *tls_enabled {
+            options
+                .ssl_root_cert_from_pem(trusted_root_certs.as_bytes().to_vec())
+                .ssl_mode(PgSslMode::VerifyFull)
+        } else {
+            options
+        };
+
+        let pool = PgPoolOptions::new()
+            .max_connections(1)
+            .min_connections(1)
+            .connect_with(options)
+            .await?;
```

Comment:
If we hardcode this params, could you lift them into constants?

Reply:
Done.